math/hyper.c

```diff
@@ -14,29 +14,61 @@
  */
 
 #include <math.h>
+#include <float.h>
+#include <errno.h>
 
 
 double cosh(double x)
 {
-	if (x == INFINITY || x == -INFINITY)
+	double y;
+
+	if (isnan(x) != 0) {
+		errno = EDOM;
+		return NAN;
+	}
+
+	if ((x == INFINITY) || (x == -INFINITY)) {
 		return INFINITY;
+	}
 
-	return (exp(x) + exp(-x)) / 2;
+	/* Make sure exp(x) is not infinity */
+	if (x < 709.78) {
```

Comment:
It's 709.79 in exp.c:198, which value is correct one?

Please mind that 709.78 < 709.7899999 < 709.79

Reply:
The value of the argument at which exp(x) evaluates to INFINITY is approximately 709.7825. However, this is not defined as a constant in any standard, so I use a safe margin from this value.